runtime/query.go

```diff
@@ -158,6 +141,24 @@ func (r *Runtime) Query(ctx context.Context, instanceID string, query Query, pri
 	return nil
 }
 
+func (r *Runtime) metricsViewCacheKey(ctx context.Context, instanceID, name string, priority int) ([]byte, error) {
+	cacheKeyResolver, err := r.Resolve(ctx, &ResolveOptions{
+		InstanceID:         instanceID,
+		Resolver:           "metrics_cache_key",
+		ResolverProperties: map[string]any{"metrics_view": name},
+		Args:               map[string]any{"priority": priority},
+		Claims:             &SecurityClaims{SkipChecks: true},
+	})
+	if err != nil {
+		return nil, err
+	}
+	cacheKey, err := cacheKeyResolver.MarshalJSON()
```

Comment:
Maybe use `.Next()` and access `res["key"]` since it's already being serialized to a string/bytes upstream? 

Or alternatively, you could consider returning `*drivers.Result` from `metricsview.Executor.CacheKey`? Then we would get support for the cache key SQL returning multiple columns.

Reply:
> Maybe use .Next() and access res["key"] since it's already being serialized to a string/bytes upstream?

Sounds good. 

> You could consider returning *drivers.Result from metricsview.Executor.CacheKey

I don't understand how this would work since the callers need to access this via `ResolverResult` interface.

> Then we would get support for the cache key SQL returning multiple columns.

Given `Executor.CacheKey` returns `[]bytes`, technically it can still support multiple columns.